### ISSUE TITLE
ReadWalkers receive unmapped reads when there is an interval list.

### ISSUE THREAD

**Issue by vruano** [broadinstitute/hellbender#430]
It seems that recently something has changed and read-walkers now receive unmapped reads even when the analysis has been restricted to some regions using the -L argument.

I presume that is only the case when their mate is mapped and in the region. 

is this change intended or accidental?

**Comment by droazen** [comment]
I believe this only happens for "unmapped" reads that have an actual coordinate on the genome -- can you confirm that this is the case? 

The underlying htsjdk method that we rely on (`SAMReader.queryOverlapping(intervals)`) will return unmapped reads only "if it has a coordinate for the purpose of sorting that is in the query region."

How do you propose that we handle these "unmapped" reads with an actual location?

**Comment by vdauwera** [comment]
Is there any way to specify a read filter that says "don't give me unmapped reads that have a location"?

**Comment by droazen** [comment]
Yes, one way to address this is via a new `ReadFilter`

**Comment by lbergelson** [comment]
It would have to be added to `WELLFORMED`

**Comment by droazen** [comment]
No, because we only want it on if intervals are not specified...

**Comment by vdauwera** [comment]
@lbergelson Not sure we want to tie this to being "well formed". Reads that are technically unmapped but placed in a general location are legal (afaik) and may be useable by some tools down the road. I would favor a mechanism by which they are selectively included/excluded. How about a separate property that would signify "well placed" instead?

**Comment by vruano** [comment]
Yup, it has the coordinates of its mapped mate. 
I figure out how to remove it from the flow of reads that apply is invoked on overwriting makeReadFilter(). 
I'm ok with this approach. It seems that some times some tools may want to see those as geraldine says. Is just that I didn't expect it. 

Will close this thread (as resolved) unless people want to add more to the discussion.

**Comment by droazen** [comment]
We should at least document this behavior better.
